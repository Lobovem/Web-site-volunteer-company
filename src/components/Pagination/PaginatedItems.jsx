import React, { useEffect, useState } from 'react';
import ReactPaginate from 'react-paginate';

// Example items, to simulate fetching from another resources.
import data from '../../data/data.json';
import s from './PaginatedItems.module.scss';
import { NewsAll } from '../NewsAll/NewsAll';

export const PaginatedItems = ({ itemsPerPage }) => {
  // Here we use item offsets; we could also use page offsets
  // following the API or data you're working with.
  const [item, setItem] = useState(0);

  // Simulate fetching items from another resources.
  // (This could be items from props; or items loaded in a local state
  // from an API endpoint with useEffect and useState)
  const endOffset = item + itemsPerPage;
  const currentItems = data.slice(item, endOffset);
  const pageCount = Math.ceil(data.length / itemsPerPage);

  // Invoke when user click to request another page.
  const handlePageClick = (event) => {
    const newOffset = (event.selected * itemsPerPage) % data.length;
    setItem(newOffset);
  };

  return (
    <>
      <NewsAll currentItems={currentItems} />
      <ReactPaginate
        breakLabel="..."
        nextLabel=">"
        onPageChange={handlePageClick}
        marginPagesDisplayed={2}
        pageRangeDisplayed={5}
        pageCount={pageCount}
        previousLabel="<"
        nextClassName={s.pagination__btn}
        previousClassName={s.pagination__btn}
        renderOnZeroPageCount={null}
        className={s.pagination}
        pageLinkClassName={s.pagination__link}
        activeClassName={s.active}
        disabledLinkClassName={s.disabled}
        // hrefBuilder={(pageNumber, pageCount) => (pageNumber <= 3 && pageNumber <= pageCount ? `/page/${pageNumber}` : '#')}
        // hrefAllControls
      />
    </>
  );
};
