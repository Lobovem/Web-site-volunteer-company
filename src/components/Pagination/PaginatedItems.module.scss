@import './src/styles/MixinsMedia.module.scss';
@import './src/styles/Variables.module.scss';
@import './src/styles/MixinsLayout.module.scss';
@import './src/styles/MixinsKit.module.scss';


.pagination {
  display: flex;
  gap: 10px;
  justify-content: center;
  align-items: center;
  list-style-type: none;
  font-weight: $font-weight-400;
  font-size: $font-size-18;
  line-height: 18px;
  color: $color-black;
  padding: 0;
  margin: 0;

  &__btn {
    color: $color-main;
    font-size: 26px;
    cursor: pointer;
    margin: 0 10px;
  }

  &__link {
    cursor: pointer;
    width: 34px;
    height: 34px;
    display: flex;
    justify-content: center;
    align-items: center;

  }
}

.active {
  background: $color-main;
  color: $color-white;
}

.disabled {
  color: $color-grey-dark;
}

.news {


  &__wrap {
    display: flex;
    gap: 40px;
    flex-direction: column;
  }

  &__inner {
    display: flex;
    flex-direction: column;
    align-items: center;

    @media (min-width:750px) {
      gap: 20px;
      flex-direction: row;
      align-items: start;

    }

    @include lg-min {
      gap: 60px;
    }

    @include xl-min {
      gap: 100px;
    }
  }

  &__imgWrap {
    margin-bottom: 20px;
    justify-content: center;
  }

  &__img {
    max-width: 360px;
    border-radius: 15px;
  }

  &__textBox {
    display: flex;
    flex-direction: column;
    gap: 20px;

    @media (min-width:750px) {
      gap: 40px;
    }

  }

  &__textWrap {
    position: relative;
    max-height: 220px;
    overflow: hidden;

    &:before {
      position: absolute;
      right: 0;
      bottom: 0;
      left: 0;
      height: 100px;
      content: "";
      opacity: 1;
      background: -webkit-gradient(linear, left top, left bottom, from(rgba(255, 255, 255, 0)), to(#fff));
      background: -o-linear-gradient(top, rgba(255, 255, 255, 0) 0, #fff 100%);
      background: linear-gradient(180deg, rgba(255, 255, 255, 0) 0, #fff 100%);
      -webkit-transition: opacity .25s;
      -o-transition: opacity .25s;
      transition: opacity .25s;
    }
  }

  &__desc {
    font-size: $font-size-14;
    line-height: 21px;
    color: $color-black;
    mix-blend-mode: normal;

    @include xl-min {
      margin-bottom: 50px;

    }
  }

  &__link {
    display: block;
    margin-right: auto;
    font-weight: $font-weight-600;
    font-size: $font-size-14;
    line-height: 14px;
    color: $color-main;
    mix-blend-mode: normal;
    text-decoration: none;

    @include hoverAnimation;

    &:hover {
      color: $color-hover;
    }
  }
}