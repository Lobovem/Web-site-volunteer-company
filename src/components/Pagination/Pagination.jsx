import React, { useState } from 'react';
import ReactPaginate from 'react-paginate';
import { NewsList } from '../NewsList/NewsList';
import { useSelector } from 'react-redux';
import { listNewsSelector } from '../../redux/slice/contentSlice';
import s from './Pagination.module.scss';

export const Pagination = ({ itemsPerPage }) => {
  const listNews = useSelector(listNewsSelector);

  // Here we use item offsets; we could also use page offsets
  // following the API or data you're working with.
  const [item, setItem] = useState(0);

  // Simulate fetching items from another resources.
  // (This could be items from props; or items loaded in a local state
  // from an API endpoint with useEffect and useState)
  const endOffset = item + itemsPerPage;
  const currentItems = listNews.slice(item, endOffset);
  const pageCount = Math.ceil(listNews.length / itemsPerPage);

  // Invoke when user click to request another page.
  const handlePageClick = (event) => {
    const newOffset = (event.selected * itemsPerPage) % listNews.length;
    setItem(newOffset);
  };

  return (
    <>
      <NewsList currentItems={currentItems} />
      <ReactPaginate
        breakLabel="..."
        nextLabel=">"
        onPageChange={handlePageClick}
        marginPagesDisplayed={2}
        pageRangeDisplayed={5}
        pageCount={pageCount}
        previousLabel="<"
        nextClassName={s.pagination__btn}
        previousClassName={s.pagination__btn}
        renderOnZeroPageCount={null}
        className={s.pagination}
        pageLinkClassName={s.pagination__link}
        activeClassName={s.active}
        disabledLinkClassName={s.disabled}
        onClick={() => window.scrollTo({ top: 0, behavior: 'smooth' })}
        // hrefBuilder={(pageNumber, pageCount) => (pageNumber <= 3 && pageNumber <= pageCount ? `/page/${pageNumber}` : '#')}
        // hrefAllControls
      />
    </>
  );
};
