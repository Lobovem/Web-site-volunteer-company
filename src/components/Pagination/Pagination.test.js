import { render, screen } from '@testing-library/react';
import { Pagination } from './Pagination';
import { useSelector } from 'react-redux';

jest.mock('react-redux', () => {
  const originalModule = jest.requireActual('react-redux');
  return {
    ...originalModule,
    useDispatch: jest.fn(),
    useSelector: jest.fn(),
  };
});

describe('Pagination', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  const data = [{ id: 1, title: 'Новина', desc: 'News' }];

  test('renders Pagination component', () => {
    useSelector.mockReturnValue(data);

    const { container } = render(<Pagination />);
    expect(container).toMatchSnapshot();
  });

  test('search tag h3 (decor)', () => {
    useSelector.mockReturnValue(data);

    render(<Pagination />);
    const h3Element = screen.getByRole('heading', { level: 3 });
    expect(h3Element).toBeInTheDocument();
  });

  test('search link btn (2 item)', () => {
    useSelector.mockReturnValue(data);

    render(<Pagination />);
    const linkBtn = screen.getAllByRole('button');
    expect(linkBtn).toHaveLength(2);
  });
});
