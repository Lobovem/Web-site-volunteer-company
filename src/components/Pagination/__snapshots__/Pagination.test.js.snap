// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders Pagination component 1`] = `
<div>
  <div
    class="news"
  >
    <h3
      class="news__decor"
    >
      Новини
    </h3>
    <div
      class="news__wrap"
    />
  </div>
  <ul
    aria-label="Pagination"
    class="pagination"
    role="navigation"
  >
    <li
      class="pagination__btn disabled"
    >
      <a
        aria-disabled="true"
        aria-label="Previous page"
        class=" disabled"
        rel="prev"
        role="button"
        tabindex="-1"
      >
        &lt;
      </a>
    </li>
    <li
      class="pagination__btn"
    >
      <a
        aria-disabled="false"
        aria-label="Next page"
        class=""
        rel="next"
        role="button"
        tabindex="0"
      >
        &gt;
      </a>
    </li>
  </ul>
</div>
`;
